src/main/java/hudson/remoting/Engine.java

```diff
@@ -332,6 +335,37 @@ public synchronized void startEngine() throws IOException {
         }
     }
 
+    /*package*/boolean loadEngineInstrumentationListeners() {
+        boolean success = true;
+        String listenerCanonicalNamesProp = System.getProperty("hudson.remoting.Engine.engineInstrumentationListenerCanonicalNames", "");
+        if (listenerCanonicalNamesProp.trim().equals("")) return success;
+
+        String[] listenerCanonicalNames = listenerCanonicalNamesProp.split(":");
+        for (String listenerName : listenerCanonicalNames) {
+            try {
+                Class<?> clazz = Thread.currentThread().getContextClassLoader().loadClass(listenerName);
+                Object listener = clazz.getDeclaredConstructor(null).newInstance();
+                if (listener instanceof EngineInstrumentationListener) {
+                    instrumentation.add((EngineInstrumentationListener) listener);
+                    LOGGER.log(Level.INFO, String.format("Loaded an external engine instrumentation listener (%s)", listenerName));
+                } else {
+                    LOGGER.log(Level.WARNING, String.format("Failed to use %s. Not an instance of %s", listenerName, EngineInstrumentationListener.class.getCanonicalName()));
+                    success = false;
+                }
```

Comment:
Java’s built-in CCEs in newer versions are better than any message you come up with.

```suggestion
                instrumentation.add((EngineInstrumentationListener) listener);
                LOGGER.log(Level.INFO, "Loaded an external engine instrumentation listener ({0})", listenerName);
```